### PR TITLE
feat: adding support for arbitrary field

### PR DESCRIPTION
**Pull request by hthieu1110**
Related to this [PR](https://github.com/gnolang/gno/pull/1983) and  [Manfred's idea](https://github.com/gnolang/gno/pull/1983#discussion_r1644292721), this PR aims at adding possibility to add arbitrary field to profile.

I keep the same fields list defined in the struct, they would be served as "standard/base" fields and all other arbitrary fields are considered as "extra".

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: examples/gno.land/r/demo/profile/profile.gno [hunk 1]
```diff
@@ -53,42 +53,49 @@ var boolFields = map[string]bool{
...
 
-func SetStringField(field, value string) error {
+func SetStringField(field, value string) bool {
 	addr := std.PrevRealm().Addr()
-	if _, ok := stringFields[field]; !ok {
-		return errors.New("invalid string field")
+	key := addr.String() + ":" + field
+	updated := fields.Set(key, value)
+
+	event := "ProfileFieldCreated"
```
Comment (zivkovicmilos) [inline examples/gno.land/r/demo/profile/profile.gno:61]: These event keys should be constants, along with some keys

File: examples/gno.land/r/demo/profile/profile.gno [hunk 2]
```diff
@@ -53,42 +53,49 @@ var boolFields = map[string]bool{
 
 // Setters
 
-func SetStringField(field, value string) error {
+func SetStringField(field, value string) bool {
 	addr := std.PrevRealm().Addr()
-	if _, ok := stringFields[field]; !ok {
-		return errors.New("invalid string field")
+	key := addr.String() + ":" + field
```
Comment (zivkovicmilos) [inline examples/gno.land/r/demo/profile/profile.gno:58]: Why did you remove the allowed fields check?

**Review by zivkovicmilos — CHANGES_REQUESTED** [review]
Thank you for the tweak 🙏 

I've left a few comments that should be addressed before we move forward

File: examples/gno.land/r/demo/profile/profile.gno (hunk 2 above)
Comment (n0izn0iz) [reply examples/gno.land/r/demo/profile/profile.gno:58]: the original intent of this realm was to allow arbitrary fields

> The goal is to pack the contract with default fields that are the expected most common ones for customization (profile_picture, name, primary_color, is_dev (to enable more debug), etc. But the contract should support a way to set new fields, basically a key/value store associated with a profile.

source: https://github.com/gnolang/gno/pull/181#issue-1229907667

I don't know why the check are there in the first place, we looked through the conversations in the initial PR and we found no rationale for these checks

File: examples/gno.land/r/demo/profile/profile.gno (hunk 1 above)
Comment (hthieu1110) [reply examples/gno.land/r/demo/profile/profile.gno]: thanks @zivkovicmilos  for your review. I've updated the code including some minor refactoring to use constants instead of hardcoded strings for other places

**Comment by zivkovicmilos** [comment]
@hthieu1110 
Can you please check why the CI is failing?

**Review by zivkovicmilos — APPROVED** [review]
Thank you for applying the fixes 🙏
